tests/test_apt.py

```diff
@@ -0,0 +1,57 @@
+'''Define unit tests for parsing APT templates with pytest.
+
+Authors
+-------
+    - Lauren Chambers
+
+Use
+---
+    Ensure you have pytest installed. Then, simply run pytest in any
+    parent directory of nircam_simulator/tests/:
+    >>> pytest
+'''
+
+import os
+
+from nircam_simulator.scripts import write_observationlist, yaml_generator, utils
+from nircam_simulator.scripts.get_catalog import get_all_catalogs
+
+INSTRUMENTS = ['NIRCam']
+PROPOSAL_ID = '1111'
+
+TESTS_DIR = os.path.realpath(os.path.join(os.getcwd(), os.path.dirname(__file__)))
+
+
+def test_APT():
+    '''Parse the given APT files and create a set of .yamls for each
+    instrument in the INSTRUMENTS list.
+    '''
+    for instrument in INSTRUMENTS:
+        # Define .pointing and .xml file locations
+        pointing_file = os.path.join(TESTS_DIR, 'test_data', instrument + 'Test.pointing')
+        xml_file = os.path.join(TESTS_DIR, 'test_data', instrument + 'Test.xml')
+
+        # Point to appropriate output directory
+        out_dir = os.path.join(TESTS_DIR, 'test_data', 'APT_{}_out'.format(instrument))
+
+        # Create/locate catalogs for target(s)
+        sw_cats, lw_cats = get_all_catalogs(pointing_file, PROPOSAL_ID)
+
+        # Write observationlist.yaml
+        observationlist_file = os.path.join(out_dir, instrument + '_observationlist.yaml')
+        write_observationlist.write_yaml(xml_file, pointing_file, observationlist_file,
+                                         ps_cat_sw=sw_cats, ps_cat_lw=lw_cats)
+
+        # Create a series of data simulator input yaml files
+        yam = yaml_generator.SimInput()
+        yam.input_xml = xml_file
+        yam.pointing_file = pointing_file
+        yam.siaf = utils.get_siaf()['NIRCam']
+        yam.output_dir = out_dir
+        yam.simdata_output_dir = out_dir
+        yam.observation_table = observationlist_file
+        yam.use_JWST_pipeline = False
+        yam.use_linearized_darks = True
+        yam.datatype = 'linear'
+        yam.reffile_setup()
+        yam.create_inputs()
```

Comment:
Since this creates a lot of temporary files that are only used for the test, one clean solution is to use the pytest yield function, which allows you to automatically clean up temporary files afterwards. This is just a suggestion and can find an example in the tests I wrote for jwql.permissions (https://github.com/spacetelescope/jwql/blob/master/jwql/tests/test_permissions.py).

Reply:
That sounds like a great idea, and is definitely something I would like to implement. However, from looking over your example and trying to find other examples online, I am kind of confused as to how I would do so in this case. (Decorators make things so much harder to understand, and I haven't been able to find a helpful explanation of what `yield` does...) If you're up for it, I would be happy to meet at some point and go over how this could be implemented?